pkg/discovery/discovery.go

```diff
@@ -124,7 +128,55 @@ func Run(kubeClient kubernetes.Interface, cfg *config.Config) (errCount int) {
 		defer os.RemoveAll(outpath)
 	}
 	trackErrorsFor("assembling results tarball")(err)
+
+	// 9. Mark final annotation stating the results are available and status is completed.
+	trackErrorsFor("updating pod status")(
+		updateStatus(kubeClient, cfg.Namespace, pluginaggregation.CompleteStatus),
+	)
+
 	logrus.Infof("Results available at %v", tb)
 
 	return errCount
 }
+
+// updateStatus changes the summary status of the sonobuoy pod in order to
+// effect the finalized status the user sees. This does not change the status
+// of individual plugins.
+func updateStatus(client kubernetes.Interface, namespace string, s string) error {
```

Comment:
This was the main crux of the work here and I tried a few different things.
1) I thought about moving some of the status/update code up out of the aggregator but as I started doing that I found that it ends up hitting the main interface that was exported to external consumers so I thought I should just leave it where it is.
2) I thought about referencing the `pkg/client` code to get the status (which looks very similar to this) but I wasn't sure if we wanted to tie the `discovery` and `client` packages together in the first place. Secondly, the client uses a `SonobuoyClient` and we just have the `kubernetes.Interface` available here.
3) I tried just patching the one status field that I want to edit, but unfortunately the entire annotation is just a json encoded string and not an object which can be patched. As a result I currently have to get the whole thing, unmarshal it, modify it, and repackage it. This means pulling in those new imports into this package.

I'd be happy to swap this out somehow for something simpler, but as of now it seems like that patch isn't possible without reading and unmarshaling the data first which requires all this extra logic.